kratos/solving_strategies/strategies/line_search_strategy.h

```diff
@@ -274,56 +323,125 @@ class LineSearchStrategy
+                TSparseSpace::Assign(aux,x, Dx);
+            } else {
+                if(std::abs(r1)>std::abs(r2)) {
+                    r1 = rf;
+                    x1 = x;
+                } else {
+                    r2 = r1;
+                    x2 = x1;
+                    r1 = rf;
+                    x1 = x;
+                }
+                converged = false;
+            }
+
+
+            it++;
+        }
         TSparseSpace::SetToZero(b);
-        pBuilderAndSolver->BuildRHS(pScheme, BaseType::GetModelPart(), b );
-        double rh = TSparseSpace::TwoNorm(b);
+    }
 
-        //compute full step residual (add another half Dx to the previous half)
-        BaseType::UpdateDatabase(A,aux,b,MoveMesh);
-        TSparseSpace::SetToZero(b);
-        pBuilderAndSolver->BuildRHS(pScheme, BaseType::GetModelPart(), b );
-        double rf = TSparseSpace::TwoNorm(b);
-
-        //compute optimal (limited to the range 0-1)
-        //parabola is y = a*x^2 + b*x + c -> min/max for
-        //x=0   --> r=ro
-        //x=1/2 --> r=rh
-        //x=1   --> r =
-        //c= ro,     b= 4*rh -rf -3*ro,  a= 2*rf - 4*rh + 2*ro
-        //max found if a>0 at the position  xmax = (rf/4 - rh)/(rf - 2*rh);
-        double parabola_a = 2*rf + 2*ro - 4*rh;
-        double parabola_b = 4*rh - rf - 3*ro;
-        double xmin = 1e-3;
-        double xmax = 1.0;
-        if( parabola_a > 0) //if parabola has a local minima
-        {
-            xmax = -0.5 * parabola_b/parabola_a; // -b / 2a
-            if( xmax > 1.0)
-                xmax = 1.0;
-            else if(xmax < -1.0)
-                xmax = -1.0;
-        }
-        else //parabola degenerates to either a line or to have a local max. best solution on either extreme
-        {
-            if(rf < ro)
-                xmax = 1.0;
-            else
-                xmax = xmin; //should be zero, but otherwise it will stagnate
-        }
+    /**
+     * @brief This method returns the default settings
+     */
+    Parameters GetDefaultSettings() override
+    {
+        Parameters base_default_settings = BaseType::GetDefaultSettings();
+        Parameters default_settings(R"({
+            "max_line_search_iterations" : 5,
+            "first_alpha_value"          : 0.5,
+            "second_alpha_value"         : 1.0,
+            "min_alpha"                  : 0.1,
+            "max_alpha"                  : 2.0,
+            "line_search_tolerance"      : 0.5
+        })");
+        default_settings.AddMissingParameters(base_default_settings);
+        return default_settings;    
+    }
 
-        //perform final update
-        TSparseSpace::Assign(aux,-(1.0-xmax), Dx);
-        BaseType::UpdateDatabase(A,aux,b,MoveMesh);
+    /**
+     * @brief This method assigns settings to member variables
+     */
+    void AssignSettings() override
+    {
+        BaseType::AssignSettings();
+        mMaxLineSearchIterations = this->mSettings["max_line_search_iterations"].GetInt();
```

Comment:
you miss validating the parameters. note that you can validate in the derived class against the derived class settings, since in there you already inherited from the base class ones

Reply:
they are validated in the constructor